src/app/(public-pages)/deals/add/actions.tsx

```diff
@@ -28,26 +29,37 @@ export const submitDealAction = async (
 
   const validated = newDealSchema.safeParse(dealFromForm)
 
+
+
+
   if (validated.success) {
+    const createDealAndSendEmails = async () => {
```

Comment:
I don't think we need to create a nested function here. Could we call this logic directly?

Reply:
I did this to make the nesting of the try/catch cleaner (it is still nested but looks a little cleaner this way).  I wanted to be able create a deal and also know when an email fails after creation. This will properly show in Inngest when the email fails.  It also shows in Inngest when a deal creation fails.